### PR TITLE
Multi mode

### PR DESCRIPTION
**Pull request by gdozot2**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,8 +225,8 @@ To activate this mode, add in your HTML code the web component with the followin
 For this scenario, there is one events to listen:
 
 - `position-selected`: This event is sent in two cases:
-  - When all the rules are met and the position is available. The information is stored in an object in event.detail and this object contains the id of the object and the geometry in WTK format.
-    - { id: 54, geometry: POINT (2538545.7462747833 1180732.9753953428)) };
+  - When all the rules are met and the position is available. The information is stored in an array of object in event.detail and this object contains only the geometry in WTK format.
+    - [{ id: 54, geometry: POINT (2538545.7462747833 1180732.9753953428)) }];
```
Comment (slecorne) [inline README.md]: in that scenario there is the id of the object also, no?

File: src/components/notification-manager.ts [hunk 2]
```diff
@@ -66,78 +69,55 @@ export default class NotificationManager {
...
-                const geometry = {
-                    type: "Point",
-                    coordinates: feature.get('geometry').getCoordinates()
-                  };
-                GeocityEvent.sendEvent('position-selected', {
-                    id: feature.get('name'),
-                    geometry: wtk.stringify(geometry)
-                });
+            const features = useStore().getSelectedFeatures();
+            console.log(features)
```
Comment (slecorne) [inline src/components/notification-manager.ts]: remove console.log

File: src/components/notification-manager.ts [hunk 3]
```diff
@@ -66,78 +69,55 @@ export default class NotificationManager {
...
-                    id: feature.get('name'),
-                    geometry: wtk.stringify(geometry)
-                });
+            const features = useStore().getSelectedFeatures();
+            this.checkMaxElementContraint(features);
+            if (this.validZoomConstraint && this.validMaxElementConstraint && features.length > 0) {
+                GeocityEvent.sendEvent('position-selected', this.generateExportData(features));
+            }
+            else {
+                GeocityEvent.sendEvent('position-selected', undefined);
```
Comment (slecorne) [inline src/components/notification-manager.ts]: we should not reset all the selection in the case more than max elements are selected. We should probably only prevent the selection/creation of a new element

File: src/components/notification-manager.ts [hunk 4]
```diff
@@ -66,78 +69,55 @@ export default class NotificationManager {
...
                 } else {
-                    const geometry = {
-                        type: "Point",
-                        coordinates: feature.get('geom').getCoordinates()
-                      };
-                    GeocityEvent.sendEvent('position-selected', {
-                        id: feature.get('objectid'),
-                        geometry: wtk.stringify(geometry)
-                    });
+                    GeocityEvent.sendEvent('position-selected', undefined);
```
Comment (slecorne) [inline src/components/notification-manager.ts]: we should not reset all the selection in that case. We should probably only prevent the selection/creation of a new element

File: src/components/single-create.ts [hunk 5]
```diff
@@ -130,9 +155,16 @@ export default class SingleCreate {
...
+          id: Number(`${Math.round(coordiante[0])}${Math.round(coordiante[1])}`),
+          isSelected: true
         });
-        useStore().setSelectedFeature(feature)
+        feature.setGeometryName('geom');
+        if (useStore().getMaxElement() === 1) {
+          useStore().removeSelectedFeature(useStore().getCurrentItemId(), 'id');
+        }
+        useStore().setCurrentItemId(feature.get('id'))
+        useStore().addSelectedFeature(feature)
```
Comment (slecorne) [inline src/components/single-create.ts]: may be at some point would be good to refactor with a class variable `store = useStore()` for readability

File: src/components/single-select.ts [hunk 6]
```diff
@@ -30,15 +30,17 @@ export default class SingleSelect {
...
+            }
+            GeocityEvent.sendEvent('icon-clicked', feature.getProperties().features[0].get('objectid'));                
           }
         }
       });
     });
 
     window.addEventListener('recenter-selected-element', () => {
-      useStore().getMap().getView().setCenter(useStore().getSelectedFeature()?.get('geom').getCoordinates())
+      useStore().getMap().getView().setCenter(useStore().getSelectedFeature(useStore().getCurrentItemId(), 'objectid')?.get('geom').getCoordinates())
```
Comment (slecorne) [inline src/components/single-select.ts]: for readability
```suggestion
const currentItemID = store.getCurrentItemId();
const coords = store.getSelectedFeature(currentItemID, 'objectid')?.get('geom').getCoordinates();
store.getMap().getView().setCenter(coords);
```

**Review by slecorne — CHANGES_REQUESTED** [review]

File: src/components/notification-manager.ts (hunk 3 above)
Comment (gdozot2) [reply src/components/notification-manager.ts]: This implementation is based on discussion with Jonathan. You can create/select as much item as you want without prevention (except for the 1 size). But if you number of selection/creation is to high, you must delete elements until selection/creation number of elements match the rule.

File: src/components/notification-manager.ts (hunk 4 above)
Comment (gdozot2) [reply src/components/notification-manager.ts]: This implementation is based on discussion with Jonathan. You can create/select as much item as you want without prevention (except for the 1 size). But if you number of selection/creation is to high, you must delete elements until selection/creation number of elements match the rule.

File: README.md (hunk 1 above)
Comment (gdozot2) [reply README.md]: done

File: src/components/notification-manager.ts (hunk 4 above)
Comment (slecorne) [reply src/components/notification-manager.ts]: I quickly checked with @jfavlam and I think there was a misunderstanding. You should not be able to select a 4th one if the max number is 3 (and display the warning). And then you need indeed to deselect one of the previous one to be able to select the new one.

File: src/components/notification-manager.ts (hunk 3 above)
Comment (gdozot2) [reply src/components/notification-manager.ts]: change done

File: src/components/notification-manager.ts (hunk 4 above)
Comment (gdozot2) [reply src/components/notification-manager.ts]: Change done

File: src/components/notification-manager.ts [hunk 7]
```diff
@@ -207,8 +199,9 @@ export default class NotificationManager {
 
     checkMaxElementContraint(features: Array<Feature>) {
         if (useStore().getMaxElement() >= 0) {
-            if (features.length > useStore().getMaxElement()) {
-                this.validMaxElementConstraint = false;
+            if (features.length >= useStore().getMaxElement()) {
+                if (features.length > useStore().getMaxElement())
+                    this.validMaxElementConstraint = false;
                 this.maxElementNotificationControl?.show();
```
Comment (slecorne) [inline src/components/notification-manager.ts]: I think I would show the warning only if trying to select an additional one. Else the warning is not so visible once the select does not work
To check with Jonathan

Comment (gdozot2) [reply src/components/notification-manager.ts]: done

**Review by slecorne — APPROVED** [review]

**Pull request merged**